networkx/algorithms/regular.py

```diff
@@ -32,6 +43,8 @@ def is_regular(G):
         d_out = G.out_degree(n1)
         out_regular = all(d_out == d for _, d in G.out_degree)
         return in_regular and out_regular
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept("G has no nodes")
```

Comment:
The decision to change the behavior here will need a bit of careful attention - is the concept of "regularity" truly nonsensical for an empty graph, or should this return False instead? @dschult @MridulS what do you think?

Currently, `nx.is_regular` on an empty graph raises a `StopIteration`, so IMO either of the alternatives would be an improvement. The question is what should be done.

Finally, for a change in behavior like this we'd also want to add a test to the test suite to verify that the behavior is as expected. In this case the value of adding a test is illustrative because you would find that `is_regular` is still giving a `StopIteration` instead of the intended `NetworkXPointlestConcept`! This is because the `StopIteration` originates from the `n1 = nx.utils.arbitrary_element(G)` line.